src/main/java/de/teamlapen/vampirism/entity/ai/EntityAIUseAction.java

```diff
@@ -0,0 +1,149 @@
+    }
+
+    @Override
+    public boolean shouldExecute() {
+        if (entity.getAttackTarget() instanceof EntityPlayer && !availableActions.isEmpty()) {
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public void startExecuting() {
+        action = null;
+        cooldown = 100;
+        duration = -1;
+    }
+
+    /**
+     * Keep ticking a continuous task that has already been started,
+     * 
+     * {@link duration} > 0 action in progress,
+     * {@link duration} = 0 action will be deactivated,
+     * {@link duration} = -1 no action is active,
+     * {@link duration} <= 0 && {@link cooldown} > 0 actions on cooldown,
+     * {@link duration} <= 0 && {@link cooldown} = 0 new action will be set & activated
+     */
+    @Override
+    public void updateTask() {
+        if (duration > 0) {
+            duration--;
+            updateAction();
+        } else if (cooldown > 0) {
+            cooldown--;
+        } else {
+            newAction();
+            cooldown = action.getCooldown(entity.getLevel());
+            if (action instanceof ILastingAction) {
+                duration = ((ILastingAction<T>) action).getDuration(entity.getLevel());
+                ((ILastingAction<T>) action).activate(entity);
+            } else if (action instanceof IInstantAction) {
+                ((IInstantAction<T>) action).activate(entity);
+            }
+        }
+
+        /* calls deactivate() for {@link ILastingAction} once per action */
+        if (duration == 0) {
+            if (action instanceof ILastingAction) {
+                ((ILastingAction<T>) action).deactivate(entity);
+            }
+            duration--;
+        }
+    }
+
+    @Override
+    public boolean isInterruptible() {
+        return false;
+    }
+
+    @Override
+    public void resetTask() {
+        if (action instanceof ILastingAction) {
+            ((ILastingAction<T>) action).deactivate(entity);
+        }
+    }
+
+    /**
+     * updates action based on actiontype
+     */
+    private void updateAction() {
+        if (action instanceof ILastingAction) {
+            ((ILastingAction<T>) action).onUpdate(entity, duration);
+        }
+    }
+
+    /**
+     * sets a new random action from {@link availableActions} for the entity,
+     * otherwise a placeholder
+     */
+    private void newAction() {
+        if (rand.nextInt(1) == 0) { // TODO usable balance
+                action = availableActions.get(rand.nextInt(availableActions.size()));
+        } else {
+            action = new DefaultEntityAction() {
+                @Override
+                public int getCooldown(int level) {
+                    return (int) (100 * (1 + rand.nextFloat() * 2)); // TODO edit
+                }
+            };
+        }
+    }
+
+    /**
+     * use this method if EntityAIUseAction is used in onUpdate() in an Entity
+     */
+    public void handle() {
+        if (shouldExecute()) {
```

Comment:
Not sure, but if the attack target changes to null or a non player, this stops updating the task, but does not reset/deactivate it?